### PR TITLE
Exclude Latest Two CC-licenses from Projects [PLAT-722]

### PR DESCRIPTION
**Pull request by pattisdr**
## Purpose
https://github.com/CenterForOpenScience/osf.io/pull/8229 adds two new CC-licenses as options for both OSF projects and preprints - CC-BY-NC-ND and CC-BY-SA 4.0.  This PR removes them as options for projects, while still allowing them to be used on preprints.  Also removes these licenses from the APIv2 LicenseList endpoint.

## Changes
- Remove latest two CC-licenses from project page dropdown
- Remove latest two CC-licenses from LicenseList endpoint
- Still allow the CC-licenses to be returned under PreprintProviderLicenseList if in licenses_acceptable or no licenses_acceptable specified.
- Still allow the new CC-licenses to be selected for a PreprintProvider in the admin app
- Still allow accessing new CC-licenses under LicenseDetail APIv2

## Project Page - new CC-licenses have been removed as an option
screen shot 2018-04-12 at 7 08 27 pm

## Preprint Provider options - If a preprint has made one of the new CC-licenses an option, it can be selected
screen shot 2018-04-12 at 7 07 05 pm

## Admin App Preprint Providers create form - still has both CC-licenses as options
screen shot 2018-04-12 at 7 09 12 pm

## API v2 LicenseList endpoint, 16 licenses instead of 18 are returned - new CC licenses are excluded
screen shot 2018-04-12 at 7 08 08 pm

## Ticket

https://openscience.atlassian.net/browse/PLAT-722

### CODE REVIEW

File: api/licenses/views.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class LicenseList(JSONAPIBaseView, generics.ListAPIView, ListFilterMixin):
     ordering = ('name', )  # default ordering
 
     def get_default_queryset(self):
-        return NodeLicense.objects.all()
+        return NodeLicense.objects.exclude(license_id='CCBYNCND').exclude(license_id='CCBYSA40')
```
Comment (sloria) [inline api/licenses/views.py]: It's not really clear why these are being excluded--can we have way for the code to more clearly separate project licenses from preprint-only licenses?

One idea might be to use custom managers (see https://docs.djangoproject.com/en/1.11/topics/db/managers/#modifying-a-manager-s-initial-queryset) to automatically filter:

```python
NodeLicense.objects.project_licenses.all()  # excludes CCBYNCND and CCBYSA40
NodeLicense.objects.preprint_licenses.all()
```

Comment (pattisdr) [reply api/licenses/views.py]: that's a good idea

Comment (pattisdr) [reply api/licenses/views.py]: will wait to change this @sloria until there's consensus that this is the route we're going. This WIP PR was just in response to being asked if this was possible.

Comment (pattisdr) [reply api/licenses/views.py]: Change made @sloria

**Comment by sloria** [comment]
Refactored a bit so that the new manager methods are called on `self` instead of `NodeLicense.objects`.

**Pull request merged**